### ISSUE TITLE
SchemaNameToTopic transform question

### ISSUE THREAD

**Issue by OuesFa** [jcustenborder/kafka-connect-transform-common#83]
Hi 👋 
I'm using kafka topic to store messages with different avro schemas. Schema are stored within Confluent SR. 
I would like to send messages from the same topic to different destinations (BQ tables using Kafka Connect BQ sink) depending on the schema of the message.
The `SchemaNameToTopic` seems to be promising, I would use it to change the topic name for every message depending on its schema, then the BQ sink connector would store the message within a table with the topic name. 

Am I understanding it correctly? 

If so, is there any config usage example of this transform please?
Thanks!

**Comment by jcustenborder** [comment]
Yes, that is the intended use of `SchemaNameToTopic` when the sink connector routes output by the Connect record topic after SMTs are applied.

For routing based on the value schema, configure it as a sink SMT:

```properties
transforms=routeBySchema
transforms.routeBySchema.type=com.github.jcustenborder.kafka.connect.transform.common.SchemaNameToTopic$Value
```

Use `$Key` instead if the routing should come from the key schema:

```properties
transforms=routeBySchema
transforms.routeBySchema.type=com.github.jcustenborder.kafka.connect.transform.common.SchemaNameToTopic$Key
```

Notes:
- This SMT has no extra configuration beyond the transform `type`.
- It uses the Connect schema name (`valueSchema().name()` or `keySchema().name()`), not the Schema Registry subject or schema id.
- Make sure the resulting schema name is valid for the downstream sink/table mapping. Fully qualified Avro names often contain dots, so BigQuery may need connector-side mapping/sanitization or schema names that already match valid table names.

Closing this as answered.
